code/modules/mob/living/say.dm

```diff
@@ -93,9 +113,26 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	if(!message || message == "")
 		return
 
-	if(GLOB.in_character_filter.len && !forced)
-		if(findtext(message, config.ic_filter_regex))
-			to_chat(src, "<span class='warning'>That message contained a word prohibited in IC chat! Consider reviewing the server rules.</span>")
+	if(src.client && GLOB.in_character_filter.len && !forced)
+		var/bad_message_start = findtext(message, config.ic_filter_regex)
+		if(bad_message_start)
+			// Catch potential baddies
+			var/client/C = src.client
+			C.censoredMessageCounter++
+			if (C.censoredMessageCounter > CENSOR_MESSAGE_LIMIT)
+				to_chat(C, "<span class='userdanger'>You have exceeded the limit of [CENSOR_MESSAGE_LIMIT] censored messages in 60 seconds and have been disconnected. Administrators have been notified of this event.</span><br><span class='danger'>You may reconnect via the button in the file menu or by <b><u><a href='byond://winset?command=.reconnect'>clicking here to reconnect</a></u></b>.</span>")
+				QDEL_IN(C, 1) //to ensure they get our message before getting disconnected
+				message_admins("[key_name(src)] has been kicked for exceeding the limit of [CENSOR_MESSAGE_LIMIT] censored messages in 60 seconds.")
+			else if (C.censoredMessageCounter > CENSOR_MESSAGE_LIMIT - 5)
+				to_chat(C, "<span class='warning'>You are approaching the limit of [CENSOR_MESSAGE_LIMIT] censored messages in 60 seconds. You will be disconnected if you exceed this limit.</span>")
+			else if (C.censoredMessageCounter == 0)

```

Comment:
this will never be true because you always increment it first

Reply:
GOOD point i changed the position of that line briefly before committing one sec

Reply:
resolved in https://github.com/tgstation/tgstation/pull/44712/commits/546b661e57065fa4874f5279a209a765da8682df